hera_pspec/uvpspec_utils.py

```diff
@@ -366,3 +364,37 @@ def _conj_blpair(blpair, which='both'):
 
     return conj_blpair
 
+def _fast_lookup_blpairts(src_blpts, query_blpts):
+    """
+    Helper function to allow fast lookups of array indices for large arrays of 
+    blpair-time tuples.
+    
+    Parameters
+    ----------
+    src_blpts : array_like
+        Array of size (N, 2), containing a list of (blpair, time) couplets.
+    
+    query_blpts : array_like
+        Array of size (M, 2), containing a list of (blpair, time) couplets that 
+        you want to find the indices of in source_blpts.
+    
+    Returns
+    -------
+    blpts_idxs : array_like
+        Array of integers of size (M,), which are indices in the source_blpts 
+        array for each item in query_blpts.
+    """
+    # This function works by using a small hack -- the blpair-times are turned 
+    # into complex numbers of the form (blpair + 1.j*time), allowing numpy 
+    # array lookup functions to be used
+    src_blpts = np.array(src_blpts)
```

Comment:
perhaps you should use `np.asarray` here, as it will not make a copy of the data if it already conforms to the `array_like`